### PR TITLE
[WIP] Group activities by date

### LINKED ISSUE #116: Clearly separate dates
It should be easier to see what activity is on which day. Maybe by visual separation or by a filter.

### PR DESCRIPTION
**Pull request by maser**
Closes #116 

- [x] Group activities by date
- [ ] Do not show past dates
- [ ] Make it prettier?

### CODE REVIEW

**Comment by maser** [comment]
screenshot from 2015-07-23 23 13 23

File: app/models/event.rb [hunk 1]
```diff
@@ -66,6 +66,17 @@ def search_activities(author = nil, query_string = "", filter = "all")
   end
   alias_method :activities, :search_activities
 
+  def activities_per_day(author = nil, query_string = "", filter = "all")
+    activities = search_activities(author, query_string, filter)
+    activities.sort_by! { |activity| activity.dates.first }
+
+    activities.each_with_object(Hash.new { [] }) do |activity, grouped_by_day|
```
Comment (moonglum) [inline app/models/event.rb]: Just out of curiosity: Why are you doing a `Hash.new { [] }` here? As you are using parentheses you can just write `{ [] }`, only if you don't use the parantheses Ruby doesn't know if you want to provide a hash or a block :smile: But that's nitpicky, no blocker at all :wink:

Comment (moonglum) [inline app/models/event.rb]: But I'm confused... If I try `Hash.new { [] }` in IRB, I just get an empty hash. Why not just use `{}`?

Comment (moonglum) [inline app/models/event.rb]: Oooooh now I get it. That's clever. Did not know that :smile:

Comment (bitboxer) [inline app/models/event.rb]: I think this is a block that returns the default entry if the key is not found. And in this case the default is a new array for every lookup that has no key. That way line 75 is possible.

Comment (moonglum) [inline app/models/event.rb]: Yep :smile:

**Comment by moonglum** [comment]
Great work, @maser :smile: :shipit:

File: app/models/event.rb (hunk 1 above)
Comment (maser) [inline app/models/event.rb]: I hope it's not too clever. I find default Hash values very useful, but they are not great for readability. I have thought about using [Hash#default_proc=](http://ruby-doc.org/core-2.2.2/Hash.html#method-i-default_proc-3D).

Comment (moonglum) [inline app/models/event.rb]: It did take my some time to understand what is going on (as you can see in my stream of consciousness above :laughing:). But I think it's ok :smile:

**Comment by moonglum** [comment]
Also looking good on staging. Shall we release the :octopus: ?

**Comment by maser** [comment]
Those three words were the best part of the whole movie.

RELEASE. THE. cute octopus baby.

**Comment by moonglum** [comment]
:+1:

**Pull request merged**